dispatch-service/src/main/java/de/muenchen/oss/swim/dispatcher/application/usecase/DispatcherUseCase.java

```diff
@@ -135,7 +172,50 @@ protected void processFile(final UseCase useCase, final File file) throws FileSi
         fileSystemOutPort.tagFile(file.bucket(), file.path(), Map.of(
                 swimDispatcherProperties.getDispatchStateTagKey(),
                 swimDispatcherProperties.getDispatchedStateTagValue()));
-        // update metric
-        dispatchMeter.incrementDispatched(useCase.getName(), useCase.getDestinationBinding());
     }
+
+    /**
+     * Tag file and move to finished directory.
+     *
+     * @param useCase The use case of the file.
+     * @param file The file to finish.
+     */
+    protected void finishFile(final UseCase useCase, final File file) {
+        // finish metadata file if required and exists
+        if (useCase.isRequiresMetadata() && this.fileSystemOutPort.fileExists(file.bucket(), file.getMetadataFilePath())) {
+            this.fileHandlingHelper.finishFile(useCase, file.bucket(), file.getMetadataFilePath());
+        }
+        // finish file
+        this.fileHandlingHelper.finishFile(useCase, file.bucket(), file.path());
+    }
+
+    /**
+     * Reroute a file to another use case.
+     *
+     * @param useCase The use case of the file.
+     * @param file The file to reroute
+     * @param tags The tags of the file. Used for resolving target.
+     * @throws UseCaseException If target use case can't be resolved.
+     */
+    protected void rerouteFileToUseCase(final UseCase useCase, final File file, final Map<String, String> tags) throws UseCaseException {
+        // resolve target use case
+        final String targetUseCaseName = tags.get(ACTION_REROUTE_DESTINATION_TAG_KEY);
+        if (targetUseCaseName == null) {
+            final String message = String.format("Action reroute but no target use case found for file %s in use case %s ", file.path(), useCase.getName());
+            throw new IllegalStateException(message);
+        }
+        if (useCase.getName().equals(targetUseCaseName)) {
+            final String message = String.format("Action reroute to same use case for file %s in use case %s ", file.path(), useCase.getName());
+            throw new IllegalStateException(message);
+        }
+        final UseCase targetUseCase = swimDispatcherProperties.findUseCase(targetUseCaseName);
```

Comment:
Typo in JavaDoc of `findUseCase`: "Finde UseCase via name." (german word "finde")